src/Logger.js

```diff
@@ -54,60 +104,60 @@ class Logger {
 
     this.status(emoji.error, message, 'red');
     if (stack) {
-      this.write(stack);
+      this.write(
+        `${emoji.error} ${this.chalk['red'].bold(message)}`,
+        persistent,
+        'error'
+      );
+      this.write(stack, persistent, 'error');
     }
   }
 
-  clear() {
-    if (!this.color) {
+  status(emoji, message, color = 'gray') {
+    if (this.logLevel < 3) {
       return;
     }
 
-    while (this.lines > 0) {
-      readline.clearLine(process.stdout, 0);
-      readline.moveCursor(process.stdout, 0, -1);
-      this.lines--;
-    }
+    this.messages[0].content = this.chalk[color].bold(`${emoji}  ${message}`);
 
-    readline.cursorTo(process.stdout, 0);
-    this.statusLine = null;
+    this.writeLine(0);
   }
 
-  writeLine(line, msg) {
-    if (!this.color) {
-      return this.log(msg);
-    }
-
-    let n = this.lines - line;
-    let stdout = process.stdout;
-    readline.cursorTo(stdout, 0);
-    readline.moveCursor(stdout, 0, -n);
-    stdout.write(msg);
-    readline.clearLine(stdout, 1);
-    readline.cursorTo(stdout, 0);
-    readline.moveCursor(stdout, 0, n);
+  persistent(message) {
+    this.log(this.chalk.bold(message), true);
   }
 
-  status(emoji, message, color = 'gray') {
-    if (this.logLevel < 3) {
-      return;
-    }
-
-    let hasStatusLine = this.statusLine != null;
-    if (!hasStatusLine) {
-      this.statusLine = this.lines;
+  handleMessage(options) {
+    let message = options.message;
+    let persistent = options.persistent;
+    let emoji = options.emoji;
+
+    switch (options.messageType) {
+      case 'log':
+        this.log(message, persistent);
+        break;
+      case 'warning':
+        this.warn(message, persistent);
+        break;
+      case 'error':
+        this.error(message, persistent);
+        break;
+      case 'status':
+        this.status(emoji, message, persistent);
+        break;
+      case 'persistent':
+        this.persistent(message);
+        break;
     }
+  }
+}
 
-    this.writeLine(
-      this.statusLine,
-      this.chalk[color].bold(`${emoji}  ${message}`)
-    );
-
-    if (!hasStatusLine) {
-      process.stdout.write('\n');
-      this.lines++;
-    }
+let loggerInstance;
+function getLogger() {
+  if (!loggerInstance) {
+    loggerInstance = new Logger();
   }
+  return loggerInstance;
 }
 
-module.exports = Logger;
+module.exports = getLogger();
```

Comment:
We should probably export Logger rather than an instance, so that it's possible to inherit from `Logger` or extend it with mixins.

Reply:
I'm not sure if this is really necessary but i exported both in new improvements